romanisim/tests/test_image.py

```diff
@@ -635,3 +635,109 @@ def test_reference_file_crds_match(level):
     else:
         # level = 2
         assert (type(im) is WfiImage)
+
+
+@metrics_logger("DMS231")
+@pytest.mark.soctests
+def test_inject_source_into_image():
+    """Inject a source into an image.
+    Demonstrates DMS231.
+    """
+
+    # Set constants and metadata
+    galsim.roman.n_pix = 4088
+    flux = 10e3
+    metadata = copy.deepcopy(parameters.default_parameters_dictionary)
+    metadata['instrument']['detector'] = 'WFI07'
+    metadata['instrument']['optical_element'] = 'F158'
+    metadata['exposure']['ma_table_number'] = 1
+
+    # Establish exposure timing parameters
+    ma_table = parameters.ma_table[metadata['exposure']['ma_table_number']]
+    tij = l1.ma_table_to_tij(ma_table)
+    tbar = ramp.ma_table_to_tbar(ma_table)
+    exptime = parameters.read_time * (ma_table[-1][0] + ma_table[-1][1] - 1)
+
+    # Create catalog of sources
+    twcs = wcs.get_wcs(metadata, usecrds=False)
+    rd_sca = twcs.toWorld(galsim.PositionD(
+        galsim.roman.n_pix / 2, galsim.roman.n_pix / 2))
+    cat = catalog.make_dummy_table_catalog(
+        rd_sca, bandpasses=[metadata['instrument']['optical_element']], nobj=10)
+
+    # Brighten sources
+    cat['F158'] = [flux * f158 for f158 in cat['F158']]
+
+    # Create catalog with one source for injection
+    source_cat = cat.copy()
+    source_cat.remove_rows(slice(0, 9))
+    xpos, ypos = twcs._xy(np.radians(source_cat['ra']), np.radians(source_cat['dec']))
```

Comment:
FWIW, I think I would have just given it an assigned position and type rather than something random, just because injecting a huge galaxy is not really what we want here.  e.g., a PSF.

Reply:
Injected source position is now specifically assigned.